### PR TITLE
Linux 5.4.173 patch release 02/16/23

### PR DESCRIPTION
**Pull request by hiren-pensando**

### CODE REVIEW

File: dsc-linux-5.4.173/patches-2023-02/00_penfw.patch [hunk 1]
```diff
@@ -0,0 +1,795 @@
...
++};
+diff --git a/arch/arm64/configs/elba_defconfig b/arch/arm64/configs/elba_defconfig
+index 5e26a8129a46..f9f051a7a241 100644
+--- a/arch/arm64/configs/elba_defconfig
++++ b/arch/arm64/configs/elba_defconfig
+@@ -229,7 +229,6 @@ CONFIG_WATCHDOG=y
+ CONFIG_SOFT_WATCHDOG=y
+ CONFIG_DW_WATCHDOG=y
+ CONFIG_MFD_PENSANDO_ELBASR=y
+-CONFIG_MFD_SYSCON=y
```
Comment (jbradleysmith) [inline dsc-linux-5.4.173/patches-2023-02/00_penfw.patch:51]: You decided to include this cleanup in this patch?  Intentional?  I could go either way, but I wasn't expecting to see this after our chat.

Comment (hiren-pensando) [reply dsc-linux-5.4.173/patches-2023-02/00_penfw.patch:51]: I could either way as well. But I thought, it will keep future patches clean.

Comment (jbradleysmith) [reply dsc-linux-5.4.173/patches-2023-02/00_penfw.patch:51]: You could structure this as 2 patches, 1) 00_penfw and 2) 01_elba_defconfig_cleanup so it is clear what is penfw, but up to you.

**Review by jbradleysmith — APPROVED** [review]
lgtm

**Comment by hiren-pensando** [comment]
Separated it into 2 patches:

With that:

```
[ibm]/local3/hiren/ibm/linux-patch-work/dsc-linux-5.4-patchtest:10098% git am  /local3/hiren/ibm/linux-patch-work/dsc-linux-patches/dsc-linux-5.4.173/patches-2023-02/00_penfw.patch
Applying: drivers/soc/pensando: penfw driver
[ibm]/local3/hiren/ibm/linux-patch-work/dsc-linux-5.4-patchtest:10099% git am  /local3/hiren/ibm/linux-patch-work/dsc-linux-patches/dsc-linux-5.4.173/patches-2023-02/01_elba_defconfig_cleanup.patch
Applying: defconfig: cleanup elba_defconfig

```

**Comment by hiren-pensando** [comment]
Tested with fip.img (bl2+bl31+bl33):
```
# uname -a
Linux elba 5.4.173-00036-gdf47ed772b7e-dirty #3 SMP Thu Feb 16 18:42:04 PST 2023 aarch64 GNU/Linux

```
```
# cat /proc/cpuinfo 
processor	: 0
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 1
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 2
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 3
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 4
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 5
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 6
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 7
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 8
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 9
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 10
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 11
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 12
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 13
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 14
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 15
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0
```

```
 cd /sys/firmware/pensando/penfw/
# ls
bl1        bl31       lifecycle  pentrust
# cd bl1/
# ls
ar_nv_cntr  ar_version  image_slot  upgrade     version
# cat ar_nv_cntr 
0
# cat ar_version 
1
# cat image_slot 
0
# cat upgrade 
0
# cat version 
2
```

**Comment by hiren-pensando** [comment]
Verified that with uboot without bl31 is able to load the kernel without PSCI.  Was able to boot to kernel and all cpus working.

**Comment by hiren-pensando** [comment]
Verified with uboot with bl31 (our regular uboot) with PSCI kernel.

1. Booted successfully with all cores running.
2. penfw driver interaction was successful as shown in the below log.

```
# uname -a
Linux elba 5.4.173-00037-g673c042cc358 #1 SMP Fri Feb 17 19:23:52 PST 2023 aarch64 GNU/Linux
# cat /proc/cpuinfo 
processor	: 0
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 1
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 2
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 3
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 4
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 5
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 6
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 7
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 8
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 9
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 10
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 11
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 12
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 13
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 14
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

processor	: 15
BogoMIPS	: 400.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd08
CPU revision	: 0

# cd /sys/firmware/pensando/penfw/
# find . -print
.
./bl31
./bl31/sw_version
./pentrust
./pentrust/image_slot
./pentrust/upgrade
./pentrust/version
./lifecycle
./lifecycle/next_state
./lifecycle/boot_state
./bl1
./bl1/image_slot
./bl1/upgrade
./bl1/version
./bl1/ar_version
./bl1/ar_nv_cntr
# cd bl1
# cat version 
2
# cat ar_version 
0
# cat ar_nv_cntr 
0
# cat upgrade 
0
# cat image_slot 
0
# cd ../pentrust/
# cat image_slot 
0
# cat version 
2
# cat upgrade
0
# cd ../bl31/
# cat sw_version 
v2.4(release):3793de7ffa
# 

```

**Review by dac2 — APPROVED** [review]
LGTM

**Pull request merged**